// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package gull;

service Checker {
    rpc CheckWorkStatus (WorkRequest) returns (stream WorkReply) {}
}

service Data {
    rpc GetData (DataRequest) returns (stream DataReply) {}
}

message WorkRequest {
    string workname = 1;
}

message WorkReply {
    string workname = 1;
    string jobstatus = 2;
    string percent = 3;
}

message DataRequest {
    string workname = 1;
}

message DataReply {
    string data = 1;
}
